src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -218,12 +219,21 @@ def logout_all_users(self):
         for e in file_extensions.values():
             _try_remove(self._token_cache_file + e)
 
-    def logout_service_principal(self, sp):
-        # remove service principal secrets
-        self._service_principal_store.remove_entry(sp)
+    def logout_service_principal(self, client_id):
+        # If client_id is a username, it is ignored
+
+        # Step 1: Remove SP from MSAL token cache
+        # Note that removing SP access tokens shouldn't rely on SP store
+        cca = ConfidentialClientApplication(client_id, **self._msal_app_kwargs)
+        cca.remove_tokens_for_client()
+
+        # Step 2: Remove SP from SP store
+        self._service_principal_store.remove_entry(client_id)
```

Comment:
I originally wrote the code as 

```py
    def logout_service_principal(self, client_id):
        try:
            entry = self._service_principal_store.load_entry(client_id, self.tenant_id)
        except CLIError as ex:
            logger.info('%s %s is possibly a user account.', ex, client_id)
            return

        # Step 1: Remove service principal access tokens from MSAL token cache.
        sp_auth = ServicePrincipalAuth(entry)
        cred = ServicePrincipalCredential(sp_auth, **self._msal_app_kwargs)
        cred.remove_tokens_for_client()

        # Step 2: Remove service principal secrets
        self._service_principal_store.remove_entry(client_id)
```

Then I realized removing an SP from token cache doesn't require `client_credential`, so there is no need to read from `self._service_principal_store` and create a fully-functioning `ServicePrincipalCredential`.